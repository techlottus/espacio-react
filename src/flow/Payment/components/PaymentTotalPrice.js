import React, { useEffect, useState } from "react";
import { useSelector } from "react-redux";
import "./PaymentTotalPrice.scss";
import { convertMoney } from "../../../helpers/formatCurrency";

const textForPayment = "Resto por pagar";
const textForParcial = "Pago parcial";

const PaymentTotalPrice = ({
  price,
  charge = false,
  concept = 0.0,
  amount = 0.0,
  isRemaining = false,
  isToBePaid = false,
  totalParcial = 0.0,
}) => {
  const { texts } = useSelector((state) => state.texts);
  const [textRemaining, setTextRamaining] = useState(
    texts?.payment.resume.total
  );

  useEffect(() => {
    if (isRemaining) {
      setTextRamaining(textForPayment);
    } else {
      setTextRamaining(texts?.payment.resume.total);
    }
  }, [isRemaining]);

  const renderCharge = () => {
    return charge || price - concept > 0 ? (
      <>
        <>
          <div className="paymenttotalprice">
            <div className="paymenttotalpricetitle">{textRemaining}</div>
            <div className="paymenttotalpriceunit">{convertMoney(concept)}</div>
          </div>
          <div className="paymenttotalprice">
            <div className="paymenttotalpricetitle">
              {texts?.payment.resume.amount}
            </div>
            <div className="paymenttotalpriceunit">
              {convertMoney(Math.abs(concept - amount))}
            </div>
          </div>
        </>
      </>
    ) : (
      <div className="paymenttotalprice">
        <div className="paymenttotalpricetitle">{textRemaining}</div>
        <div className="paymenttotalpriceunit">{price}</div>
      </div>
    );
  };

  return (
    <>
      {isToBePaid ? (
        <>
          <div className="paymenttotalprice">
            <div className="paymenttotalpricetitle">{textForParcial}</div>
            <div
              className="paymenttotalpriceunit"
              style={{
                color: "#459A33",
              }}
            >
              {convertMoney(totalParcial)}
            </div>
          </div>
          <div className="paymenttotalprice">
            <div className="paymenttotalpricetitle">{textForPayment}</div>
            <div className="paymenttotalpriceunit">
              {convertMoney(Math.abs(totalParcial - amount))}
            </div>
          </div>
        </>
      ) : (
        renderCharge()
      )}
    </>
  );
};

export default PaymentTotalPrice;
