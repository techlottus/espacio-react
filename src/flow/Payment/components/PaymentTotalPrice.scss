@import '../../../App.scss';

.paymenttotalprice {
  margin-top: pixeltorem(12px);
  display: flex;
  font-family: 'Nunito', sans-serif;
  justify-content: space-between;

  @media only screen and (min-width: map-get($grid-breakpoints, m)) {
    margin-top: pixeltorem(16px);
    align-content: flex-end;
    display: flex;
  }

  > .paymenttotalpricetitle {
    font-size: pixeltorem(14px);
    line-height: pixeltorem(19px);

    @media only screen and (min-width: map-get($grid-breakpoints, m)) {
      width: calc(66.6666% - #{pixeltorem(24px)});
      padding-left: pixeltorem(24px);
    }
  }

  > .paymenttotalpriceunit {
    font-family: 'Nunito Bold', sans-serif;
    font-size: pixeltorem(18px);
    line-height: pixeltorem(24px);

    @media only screen and (min-width: map-get($grid-breakpoints, m)) {
      width: 33.3333%;
      display: flex;
      justify-content: center;
    }
  }
}