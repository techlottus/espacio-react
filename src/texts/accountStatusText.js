export const accountStatusText = {
  statusTitle: "Estado de cuenta y consultas de facturas",
  resumeTitle: "Resumen de adeudo",
  total: "Total",
  errorMessage:
    "Tuvimos dificultades para cargar tu información en este momento",
  emptyMessage: {
    congrats: "Felicidades, estás al día con tus pagos",
    continue: "Continúa transformando tu futuro",
  },
  images: { 
    user: "images/user.png",
    empty: "yeah.png",
  },
  btnModal: {
    delete: "Eliminar",
    cancel: "Cancelar",
  },
  btnDownload: {
    download: "Descargar Estado de cuenta",
  },
  btnInvoice: {
    checkInvoice: "Consulta tus facturas",
  },
  tableAccountStatus: [
    { title: "Clave" },
    { title: "Ciclo" },
    { title: "Periodo" },
    { title: "Adeudo" },
  ],

  periodDetailTitle: "Detalles por periodo",
  periodDetailDescription:
    "Selecciona el programa y el periodo para ver los detalles de los movimientos; se clasifican de la siguiente manera:",
  periodDetailBullets: {
    debt: "Con adeudo",
    payed: "Pagados",
    toPay: "Por pagar",
  },
  periodDetailSelects: {
    program: "Programa",
    period: "Periodo",
  },
  accountDetailPeriod: {
    title: "Periodo",
    icon: "calendar_today",
    debt: "Saldo del periodo",
  },
  accountDetailMovements: {
    title: "Movimientos",
  },
  tableInvoiceQuery: [
    { title: "Fecha de emisión" },
    { title: "Periodo" },
    { title: "Concepto" },
    { title: "Monto" },
    { title: "RFC" },
    { title: "Preview" },
    { title: "PDF" },
    { title: "XML" },
  ],
  tableRowInvoiceQuery: [
    { title: "07 Abril 2021" },
    { title: "Periodo 5" },
    { title: "Colegiatura" },
    { title: "$2405.50" },
    { title: "LRKO485767HF2" },
  ],
  accountStatusInvoiceQuery: {
    title: "Consulta de facturas",
    description:
      "Filtra por rango de fechas, selecciona el perido que deseas consultar y presiona descargar.",
  },
  btnInvoiceQuery: {
    download: "Descargar todo",
    clear: "Limpiar",
  },
  linkInvoiceQuery: "Ver más"
};
