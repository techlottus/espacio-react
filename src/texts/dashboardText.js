export const dashboardText = {
  menu: {
    aula: {
      emoji: "mortar-board",
      text: "Aula",
    },
    book: {
      emoji: "books",
      text: "Biblioteca",
    },
    helpCenter: {
      emoji: "thought-balloon",
      text: "Centro de Ayuda",
    },
    procedures: {
      emoji: "file-folder",
      text: "Trámites",
    },
    accountStatus: {
      emoji: "chart-with-upwards-trend",
      text: "Estado de Cuenta",
    },
    payment:  {
      emoji: "credit-card",
      text: "Pagos",
    },
    invoice: {
      emoji: "page-facing-up",
      text: "Facturas"
    },
    benefits: {
      emoji: "star",
      text: "Portal de Beneficios"
    },
    sede: {
      emoji: "notebook",
      text: "SeDe"
    },
    psychopedagogical: {
      emoji: "blue-heart",
      text: "Centro de Atención Psicopedagógica"
    },
    evaluateServices: {
      emoji: "ec ec-white-check-mark",
      text: "Evalúa los servicios"
    },
    registrationSubjects: {
      emoji: "ec ec-closed-book",
      text: "Inscribe tus materias"
    },
    logout: {
      emoji: "back",
      text: "Cerrar sesión",
    },
  },
  titleWelcome: "Hola",
  images: {
    cardHeader: "images/header.png",
    empty: "empty.png",
  },
  logo: {
    urlLogo: "logo.svg"
  },
  logos: "logo.svg",
  sectionDashboard: {
    titleNews: "Noticias de tu universidad",
    titleCards: "¿Qué quieres hacer hoy?",
  },
  playIcon: "play_circle_outline",
  cards: {
    one: {
      title: "Aula",
      icon: "school",
    },
    two: {
      title: "Pagos",
      icon: "credit_card",
    },
    three: {
      title: "Biblioteca",
      icon: "import_contacts",
    },
    four: {
      title: "Trámites",
      icon: "post_add",
    },
    five: {
      title: "Inscribe tus materias",
      icon: "add_circle_outline",
    },
  },
  footerText: {
    text: "©2022 Derechos reservados Lottus Education",
    privacy: "Aviso de privacidad y términos y condiciones",
  },
};