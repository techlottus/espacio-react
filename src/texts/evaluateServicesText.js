export const evaluateServicesText = {
    evaluateServicesTitle: "Evalua tus servicios digitales",
    modalSurvey: {
        title: "Evalua tu experiencia para realizar tu pago a través de Espacio UTC",
        text: "¿Que tan probable es que recomiendes los pagos en Espacio UTC?"
    }
}