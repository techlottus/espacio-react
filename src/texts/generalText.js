export const generalText = {
  modals: {
    modalExit: {
      icon: "help",
      modalTitlePayment: "¿Seguro que deseas salir de la sección de pagos?",
      modalTitleProcedure: "¿Seguro que deseas abandonar el trámite?",
      exitBtn: "Aceptar",
      cancelBtn: "Cancelar",
    },
  },
};
