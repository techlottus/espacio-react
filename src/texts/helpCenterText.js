export const helpCenterText = {
  titleDashboard: "¿Cómo te podemos ayudar?",
  images: {
    headerHelpCenter: "images/back-texture-white.png",
    helpCenterFlowBanner: "images/back-texture-white.png",
    helpCenterFlowAcademic: "images/banner-ayuda-academica.png",
    helpCenterFlowFinancial: "images/banner-ayuda-financiera.png",
    helpCenterFlowInstalations: "images/banner-ayuda-instalaciones.png",
    helpCenterFlowPsychopedagogical: "images/banner-ayuda-psicometrica.png",
    helpCenterFlowTechnical: "images/banner-ayuda-tecnica.png",
    helpCenterFlowProcedures: "images/banner-ayuda-tramites.png",
    helpCenterFaqs: "images/banner-preguntas-frecuentes.png",
  },
  faqsAccordion: [
    {
      title: "¿Cuándo inician mis clases?",
      description:
        "Deberás escanear completo, a color y por ambos lados los siguientes documentos:",
      list: [
        "Partida o acta de nacimiento apostillada.",
        "Certificado o notas de bachillerato apostillado, donde deben señalar los grados y ciclos escolares cursados.",
        "Diploma o título del nivel bachillerato.",
        "Notas o analítico de secundaria o educación básica",
        "Diploma o título del nivel secundaria",
        "Identificación oficial (Si el alumno es menor de edad, deberá presentar identificación oficial del padre o tutor.",
      ],
    },
    {
      title: "¿Cuál navegador es el indicado para acceder la aula virtual ? ",
      description:
        "Deberás escanear completo, a color y por ambos lados los siguientes documentos",
      list: [
        "Acta de nacimiento.",
        "Certificado parcial del nivel superior, donde deben señalar los grados y ciclos escolares cursados.",
        "Certificado total de estudios de nivel anterior inmediato.",
        "Título de nivel anterior inmediato.",
        "Identificación oficial.",
        "Carta poder.",
      ],
    },
    {
      title:
        "¿Hay una fecha límite para entregar mis actividades?, ¿Dónde puedo consultarla? ",
      description:
        "Deberás escanear completo, a color y por ambos lados los siguientes documentos",
      list: [
        "Acta de nacimiento.",
        "Certificado parcial del nivel superior, donde deben señalar los grados y ciclos escolares cursados.",
        "Certificado total de estudios de nivel anterior inmediato.",
        "Título de nivel anterior inmediato.",
        "Identificación oficial.",
        "Carta poder.",
      ],
    },
    {
      title: "Tuve un problema con un profesor ¿Qué puedo hacer?",
      description:
        "Deberás escanear completo, a color y por ambos lados los siguientes documentos",
      list: [
        "Acta de nacimiento.",
        "Certificado parcial del nivel superior, donde deben señalar los grados y ciclos escolares cursados.",
        "Certificado total de estudios de nivel anterior inmediato.",
        "Título de nivel anterior inmediato.",
        "Identificación oficial.",
        "Carta poder.",
      ],
    },
    {
      title: "¿Dónde puedo consultar mis calificaciones y fechas de entrega?",
      description:
        "Deberás escanear completo, a color y por ambos lados los siguientes documentos",
      list: [
        "Acta de nacimiento.",
        "Certificado parcial del nivel superior, donde deben señalar los grados y ciclos escolares cursados.",
        "Certificado total de estudios de nivel anterior inmediato.",
        "Título de nivel anterior inmediato.",
        "Identificación oficial.",
        "Carta poder.",
      ],
    },
  ],
  titleFlowFaqs: "Preguntas frecuentes",
  titleFlowTickets: "Levanta un ticket",
  arrowFormHelpCenter: {
    icon: "arrow_back",
    text: "Atrás",
  },
  modalConfirmHelpCenter: {
    success: {
      icon: "check_circle_outline",
      title: "El ticket se generó correctamente",
      firstDescription: "Tu solicitud ha sido recibida y se ha creado el ticket",
      secondDescription: "Debes conservar el No. de ticket para poder consultar el seguimiento del mismo.",
      btns: {
        next: "Ir a centro de ayuda",
      },
    },
    error: {
      img: "empty-error.png",
      title: "Ha ocurrido un error, inténtalo más tarde",
    },
  },
  modalHelpCenter: {
    title: "Verifica que la información esté correcta",
    issue: "Asunto: ",
    assignment: "Materia: ",
    professor: "Profesor: ",
    ades: "Asesor: ",
    description: "Descripción del caso: ",
    activity: "Actividad: ",
    assignmentToReport: "Material a reportar: ",
    paymentDate: 'Fecha de pago: ',
    paymentMount: 'Monto de pago: ',
    paymentMethod: 'Medio de pago: ',
    classDate: 'Fecha de la clase: '
  },
  searchResults: {
    empty: "No se han encontrado resultados para tu búsqueda",
    error: "Ha ocurrido un error, inténtalo más tarde",
  },
  directory: {
    title: "Directorio UTC",
    tableTitleOne: "Gerentes y coordinadores",
    tableTitleTwo: "Asesores educativos",
    headerTableCoordinator: [
      { title: "Nombre" },
      { title: "Área" },
      { title: "Teléfono" },
      { title: "Correo" },
    ],
    headerTableAdvisor: [
      { title: "Nombre" },
      { title: "Área" },
      { title: "Contacto" },
    ],
  },
};
