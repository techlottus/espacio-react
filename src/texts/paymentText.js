export const paymentText = {
  "stepper": {
    "titleOne": "Concepto",
    "titleTwo": "Datos fiscales",
    "titleThree": "Método de pago"
  },
  "breadcrumb": {
    "breadOne": "Home",
    "breadTwo": "Pagos",
    "icon": "home"
  },
  "invoiceType": {
    "one": "Persona física",
    "two": "Persona moral"
  },
  "addInvoice": "Agregar razón social",
  "forms": {
    "businessName": {
      "title": "*Nombre de la razón social",
      "error": "Nombre de la razón social requerido"
    },
    "rfc": {
      "title": "*RFC",
      "error": "Ingresa un RFC válido"
    },
    "zipCode": {
      "title": "*Código postal",
      "error": "El código postal es requerido"
    },
    "state": {
      "title": "*Estado o provincia"
    },
    "city": {
      "title": "*Ciudad",
      "error": "La ciudad es requerida"
    },
    "neighborhood": {
      "title": "*Colonia",
      "error": "La colonia es requerida"
    },
    "street": {
      "title": "*Calle y número",
      "error": "La calle y el número es requerido"
    },
    "curp": {
      "title": "*CURP del alumno",
      "error": "Ingresa un CURP válido"
    },
    "email": {
      "title": "Correo alterno"
    },
    "cfdiUsage": {
      "title": "*Uso de CFDI"
    },
    "regime": {
      "title": "*Régimen Fiscal"
    },
    "fullName": {
      "title": "*Nombre completo",
      "error": "Nombre requerido"
    }
  },
  "cfdiOptions": {
    "optionOne": "Pagos por servicios educativos",
    "optionsTwo": "Gastos general",
    "optionsThree": "Por definir"
  },
  "listTaxTitle": "La factura se emitará bajo la siguiente información",
  "formTaxTitle": "Completa los datos fiscales",
  "paymentTitle": "Pagos",
  "conceptTitle": "Para comenzar seleciona el concepto que deseas pagar",
  "methodsTitle": "Selecciona el método de pago seguro que más se adapte a tus necesidades",
  "invoiceText": "Deseo facturar",
  paymentMethods: [
    {
      title: "Pago con tarjeta",
      label: " - Pago en línea",
      icon: "credit_card",
      description:
        "Paga en línea con cualquier tarjeta de crédito o débito",
    },
    {
      title: "Ficha de depósito",
      label: " - Pago en efectivo",
      icon: "description",
      description:
        "Descarga aquí tu ficha de referencia, con la cual podrás pagar en Oxxo, Seven Eleven, Farmacias del Ahooro, Telecomm, o en las ventanillas de Banorte, Santander y BBVA.",
    },
  ],
  resume: {
    resumeText: "Resumen de compra",
    title: "No tienes conceptos cargados",
    description: "Selecciona un concepto para ver el resumen",
    total: "Total",
    amount: "Recargo",
    concept: "Concepto",
    payment: "Pagar",
    next: "Continuar",
    selectNotice: "Selecciona los conceptos que deseas pagar",
    imageSelectNotice: "images/utc/icon-card.svg",
    toggleNotice: "No se pueden emitir facturas después del pago",
    invoiceText: "Deseo facturar",
    feedbackNotice: "Completa tu información",
  },
  "stick": {
    "notConcepts": "No tienes conceptos cargados",
    "resumePayment": "Resumen de compra",
    "initResume": "Comienza con tu pago fácil y seguro",
    "addArt": "Agrega los artículos que desea pagar"
  },
  "cardResume": {
    "label": "WA-202180",
    "title": "Licenciatura en Diseño gráfico",
    "subtitle": "Nuevo Ciclo 21-2",
    "price": 2405.5,
    "totalTitle": "Total",
    "totalUnit": 2405
  },
  "btnModal": {
    "next": "Continuar",
    "back": "Regresar",
    "delete": "Eliminar",
    "cancel": "Cancelar",
    "accept": "Aceptar",
    "save": "Guardar",
    "exit": "Salir"
  },
  "btns": {
    "next": "Continuar",
    "back": "Atrás",
    "save": "Guardar"
  },
  "cardInvoice": {
    "address": "Dirección:",
    "email": "Correo electrónico:",
    "businessName": "Razón social:",
    "rfc": "RFC:",
    "name": "Nombre:",
    "curp": "CURP:",
    "cfdi": "Uso de CFDI:",
    "regime": "Régimen Fiscal"
  },
  "actions": {
    "download": "Descargar",
    "payment": "Pagar"
  },
  "concepts": {
    "description": "Vencimiento:",
    "delete": "Eliminar",
    "prefix": "WA"
  },
  "modals": {
    "modalPaymentMethods": {
      "title": "Verifica que tus datos sean correctos",
      "data": {
        "title": "La factura se emitirá bajo los siguientes datos fiscales",
        "rfc": "RFC:"
      },
      "paymentText": "Ingresa el monto que quieres pagar"
    },
    "modalIcon": "help",
    "modalToggleConcept": {
      "title": "No se pueden emitir facturas después de realizar el pago. ¿Deseas realizar tu pago sin solicitar factura?"
    },
    "modalConceptDelete": {
      "title": "¿Seguro que quieres eliminar este concepto?"
    },
    "modalInvoiceDelete": {
      "title": "¿Estás seguro que deseas eliminar estos datos fiscales?"
    },
    "modalInvoice": {
      "title": "¿Son correctos tus datos fiscales?"
    },
    "modalCardDelete": {
      "title": "¿Seguro que quieres eliminar esta tarjeta?"
    }
  },
  "errors": {
    "getInvoice": "Estamos teniendo problemas para procesar datos fiscales, inténtalo más tarde",
    "maxInvoices": "Solo puedes agregar 3 RFC",
    "deleteCard": "No pudimos eliminar la información de la tarjeta, por favor intenta más tarde",
    "postCreditCard": "No se han podido guardar tus datos en este momento, inténtalo más tarde",
    "getCreditCards":  "Tuvimos dificultades para cargar la información, inténtalo más tarde",
    "emptyCreditCards": "No tienes tarjetas guardadas"
  },
  "images": {
    "emptyError": "empty-error.png",
    "emptyFirst": "empty-first.png",
    "visa": "images/pay-visa.svg",
    "american": "images/pay-american.svg",
    "diners": "images/pay-diners.svg",
    "jcb": "images/pay-jcb.svg",
    "mastercard": "images/pay-mastercard.svg",
    "discover": "images/pay-discover.svg",
    "securePayment": "images/security-pay.svg",
    "unknown": "images/pay-unknown.svg"
  },
  "stagesTitles": {
    "one": "Selecciona un concepto",
    "two": "Completa tus datos fiscales",
    "three": "Selecciona un método de pago",
    "four": "Selecciona una tarjeta"
  },
  "secure": {
    "title": "Pago 100% seguro",
    "description": "Tus tarjetas se guardan de forma segura para que puedas reutilizar el método de pago."
  },
  "success":{
    "postCreditCard":"Los datos de la tarjeta fueron guardados correctamente",
    "deleteCard": "Los datos de la tarjeta fueron eliminados correctamente"
  },
  "ulaFeedbackText": "Para facturar entra al menú superior de la derecha en la opción de facturas, después de hacer tu pago"

}