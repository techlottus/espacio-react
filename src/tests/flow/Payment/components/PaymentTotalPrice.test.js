import React from "react";
import { mount } from "enzyme";
import { Provider } from "react-redux";
import "@testing-library/jest-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import PaymentTotalPrice from "../../../../flow/Payment/components/PaymentTotalPrice";
import { paymentText } from "../../../../texts/paymentText";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
window.dataLayer = {
  push: jest.fn()
}

const initState = {
  payment: {
    data: {
      quotes: [
        {
          quote: "test",
        },
      ],
    },
  },
  texts: {
    payment: {
      ...paymentText
    }
  }
};

let store = mockStore(initState);

const handleMount = (initState) => {
  store = mockStore(initState);
  return mount(
    <Provider store={store}>
      <PaymentTotalPrice />
    </Provider>
  );
};

describe("Test <PaymentTotalPrice/>", () => {
  beforeEach(() => {});

  test("should init component", () => {
    const wrapper = handleMount(initState);
    expect(wrapper.find(".paymenttotalprice").exists()).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });
});
