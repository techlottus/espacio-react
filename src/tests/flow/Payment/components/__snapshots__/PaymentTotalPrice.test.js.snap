// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <PaymentTotalPrice/> should init component 1`] = `
<div
  className="paymenttotalprice"
>
  <div
    className="paymenttotalpricetitle"
  />
  <div
    className="paymenttotalpriceunit"
  />
</div>
`;
